### ISSUE TITLE
Add a /latest/ path to docs

### ISSUE THREAD

**Issue by e-dard** [influxdata/docs.influxdata.com#340]
I think it would be useful to have an alias type path on the docs site, so that anything that wants to reference pages on the docs, but is not interested in pinning to a specific version can link to, for example:

https://docs.influxdata.com/influxdb/latest/introduction/

Currently, this will redirect to https://docs.influxdata.com/influxdb/v0.11/introduction/, but once `0.12` lands, it would obviously go to https://docs.influxdata.com/influxdb/v0.12/introduction/

I would expect this to work for any page on the site. If page `https://docs.influxdata.com/influxdb/latest/foo/` `foo` is removed in a future version then `https://docs.influxdata.com/influxdb/latest/foo/` should start to 404 when future version lands.

###### URL for relevant page?

All of them!

###### Where did you look before opening the issue?

I was inspired by [this issue](https://github.com/influxdata/influxdb/pull/4753) where the obvious thing seems to be to have the following:

```
[**InfluxDB** configuration file](https://influxdb.com/docs/latest/introduction/installation.html#generate-a-configuration-file).
```

**Comment by gunnaraasen** [comment]
Setting up `/latest/` paths that redirect to the equivalent page for the most recent version sounds like a good idea. We'll need to figure out how to set this up in Hugo.

**Comment by beckettsean** [comment]
@gunnaraasen is there a reason that would need to be done in Hugo? Seems more like a one-time CloudFront setup of a bunch of 302s with a parameterized value for `latest`. Then with each new version deployed, the value for `latest` is updated. 

Either approach will need to handle moved pages somehow, I suppose Hugo would be better for that.

**Comment by gunnaraasen** [comment]
No reason in particular that in needs to be done by Hugo. I'd have to investigate which is the easiest to set up and maintain.

The goal would always to only have the pages redirect to the equivalent page in the latest version. There will be a 404 if that page doesn't exist. I think we should discourage the use of `latest` and only use it for pages that change very infrequently anyway. In @e-dard's linked issue, the configuration page is likely never going to be moved so it would be safe to link.

**Comment by kostasb** [comment]
https://docs.influxdata.com/influxdb/latest/
https://docs.influxdata.com/telegraf/latest/
https://docs.influxdata.com/kapacitor/latest/
https://docs.influxdata.com/chronograf/latest/

1) Unfortunately Cloudfront won't do request redirects, it only forwards the URI as-is to another origin. If we were to go with an infra-based solution we would need to deploy a proxy (probably Nginx) between Cloudfront and S3. This deployment should be redundant in 2 availability zones (losing an Nginx instance would mean no access from Cloudfront to S3) with an ELB to tie them together. Too much complexity and multiple additional points of failure.

2) Hugo doesn't seem to do aliases as we need them in this use case. The alias needs to be called by the file that will undertake the "latest" role, so we would have to add alias definitions to each single page on the docs version we want to define as latest. A lot of work.

3) I chose to go with the automated docs deployment script by making some enhancements. The latest version for each product is now defined in a variable at the start of the publishdocs script (very easy to update the version). The script then builds the website as usual. Once the website is ready the script creates "latest" alias directories. So the links to the latest versions are created outside of Hugo. Custom solution, but it allows for minimal intervention with the current docs process and it is as easy as changing a variable to update a version.

Let me know of your thoughts.

**Comment by gunnaraasen** [comment]
Would it be possible to make the `/latest/` URLs into pages that redirect to the versioned URLs instead of separate pages? The latest links are meant mostly for the readme docs on the InfluxDB repo to point people towards the latest versioned doc pages.

For option 1, I don't think we'll need to worry about redundancy on the backend web server. The CDN will absorb most traffic and we can always fall back to pointing Cloudfront at S3 if the web server fails. Long term, relying on our own web server feels more maintainable than building out custom post-publishing scripts.

**Comment by kostasb** [comment]
@gunnaraasen 
The current architecture with Cloudfront+S3 does not provide enough flexibility to set up links or modify/redirect HTTP requests. If we would like to move forward with handling more complex scenarios rather than statically serving markdown pages and CSS, we will need full backend infrastructure. 

I believe that the most viable long-term solution is option 1.  To avoid dealing with unforeseen issues in the future, it should be deployed from start in HA mode with 2 backend instances running Hugo, 2 Nginx proxies (probably residing on the same instances) and an ELB facing Cloudfront. If we are going to do a doc-server + nginx setup, it will be worth to duplicate it (minimal effort at that point) in a HA setup and never worry about downtime again. This architecture is a common baseline and will survive long term changes as it can easily scale.

All of this falls under a different discussion than the initial request though: Do we want to invest time and resources at this time to make the docs website an ha&scalable&fully featured environment?

**Comment by e-dard** [comment]
In the past I have used Cloudflare for things like this. You can very easily setup page rules that will forward or redirect based on patterns in the URL. It's very simple and useful.

Of course, I don't know if we're using Cloudflare or not in front of the site :smile:

**Comment by kostasb** [comment]
Thanks Edd!

We have deployed a new production cluster for the docs website and set up "latest" link redirections in the web server. We are good now to close this issue.
